vm/ubpf_jit_x86_64.c

```diff
@@ -491,12 +491,22 @@ translate(struct ubpf_vm *vm, struct jit_state *state, char **errmsg)
     emit1(state, 0xc3); /* ret */
 
     /* Division by zero handler */
+    const char div_by_zero_fmt[] = "uBPF error: division by zero at PC %u\n";
+    // Emit format string into machine code.
+    emit_bytes(state, (void*)div_by_zero_fmt, sizeof(div_by_zero_fmt));
+
+    // Save the address of the start of the divide by zero handler.
     state->div_by_zero_loc = state->offset;
-    const char *div_by_zero_fmt = "uBPF error: division by zero at PC %u\n";
+
+    // LEA rdx, [rip - sizeof(div_by_zero_fmt) - 7]
+    emit_basic_rex(state, 1, RIP, platform_parameter_registers[1]);
+    emit1(state, 0x8d);
+    emit_modrm(state, 0, platform_parameter_registers[1], RIP);
+    emit4(state, -sizeof(div_by_zero_fmt) - (state->offset - state->div_by_zero_loc) - sizeof(uint32_t));
```

Comment:
will be more readable if you stored `(state->offset - state->div_by_zero_loc) - sizeof(uint32_t)` in a local variable called the the `instruction_length`